src/index.ts

```diff
@@ -237,33 +240,55 @@ export class ERC725 {
         );
       }
 
-      try {
-        const encodedKeyName = encodeKeyName(schema.name);
-
-        const isKeyValid = schema.key === encodedKeyName;
-
-        if (!isKeyValid) {
-          console.warn(
-            `The schema with keyName: ${schema.name} is skipped because its key hash does not match its key name (expected: ${encodedKeyName}, got: ${schema.key}).`,
-          );
+      let isKeyValid: true | false | null = null;
+      const segments: string[] = [];
+      const errors: string[] = [];
+      // Check schema item for dynamic key names. If it has, then make sure they have valid types.
+      schema.name.replace(/<(.*?)>/g, (_, item) => {
+        if (isKeyValid == null) {
+          isKeyValid = true;
         }
+        const valid = /bytes\d*|string|bool|address|u?int\d*/.test(item);
```

Comment:
I honestly prefer to have this regex has constants in one single place with a nice name to describe what they validate

Reply:
How does that improve readability? Then you can only see the expression in vscode.